contracts/btc-staking/src/staking.rs

```diff
@@ -288,17 +350,31 @@ fn handle_slashed_delegation(
     let affected_fps = DELEGATION_FPS.load(storage, staking_tx_hash.as_ref())?;
     let fps = fps();
     for fp in affected_fps {
-        fps.update(storage, &fp, height, |fp_state| {
-            let mut fp_state =
-                fp_state.ok_or(ContractError::FinalityProviderNotFound(fp.clone()))?; // should never happen
-            fp_state.power = fp_state.power.saturating_sub(btc_del.total_sat);
-            Ok::<_, ContractError>(fp_state)
-        })?;
+        let mut fp_state = fps.load(storage, &fp)?;
+        fp_state.power = fp_state.power.saturating_sub(btc_del.total_sat);
+
+
+        // Distribution alignment
```

Comment:
Same, is it possible to define various functions for handling point alignments / reward per stake things